src/java/org/apache/cassandra/service/DataResolver.java

```diff
@@ -723,6 +725,10 @@ public UnfilteredPartitionIterator moreContents()
             ColumnFamilyStore.metricsFor(command.metadata().cfId).shortReadProtectionRequests.mark();
             Tracing.trace("Requesting {} extra rows from {} for short read protection", toQuery, source);
 
+            // If we've arrived here, all responses have been consumed, and we're about to request more. Before that
+            // happens, clear the accumulator and allow garbage collection to free the resources they used.
+            preFetchCallback.run();
```

Comment:
Being the comment so specific about what the callback does, perhaps we could also be equally specific when naming it, perhaps something like `clearResources` instead of `preFetchCallback`. WDYT?

Reply:
Yeah, I'll either do that or make it more generic, with this specific comment moving further up to where we instantiate `ShortReadPartitionsProtection`.